### DISCUSSION TITLE
Best way to configure Laravel to use HTTPS?

### DISCUSSION THREAD

**Discussion by Ali-Shaikh** [laravel/framework#34368] [General]
```public function boot(UrlGenerator $url)
    {
        if (env('ENFORCE_SSL', false)) {
            $url->forceScheme('https');
        }
        Paginator::useBootstrap();
    }
```

Currently, I am using the above to force Laravel to use HTTPS. I am using AWS Elastic Beanstalk to host my site and using Cloudflare to handle DNS and SSL. I have also set up NGINX to redirect HTTP to HTTPS 

```
if ($http_x_forwarded_proto = 'http') {
   return 301 https://$host$request_uri;
}
```

and then I ran into Invalid Signature issue 'Email verification 403 invalid signature' which I resolved by 

```/**
     * The trusted proxies for this application.
     *
     * @var array|string|null
     */
    protected $proxies = '*';
```

I want to ask is something I am doing redundant or wrong? How can I improve my process and if there is a better way to achieve https and SSL in Laravel.

Thanks in advanced!

**Comment by jdecode** [comment]
I have always used the following solution:

In `AppServiceProvider`
Method name : `boot`
```
if($this->app->environment('production')) {
    \URL::forceScheme('https');
}
```
Source : https://stackoverflow.com/a/51819095/867451

And use SSL links in views/routes
